python/polycap.pyx

```diff
@@ -544,18 +1018,100 @@ cdef class Photon:
 
         return weights_np
 
+    @property
+    def extleaks(self):
+        '''Retrieve exterior :ref:``Leak`` class array from a :ref:``Photon`` class '''
+        if self.photon is NULL:
+            return None
+
+        cdef polycap_error *error = NULL
+
+        if self.n_ext_leaks == 0:
+            polycap_photon_get_extleak_data(self.photon, &self.ext_leaks, &self.n_ext_leaks, &error)
+            polycap_set_exception(error)
+
+        if self.n_ext_leaks > 0:
+            for i in range(self.n_ext_leaks):
+                leak = Leak.create(self.ext_leaks[i])
+                yield leak
+        else:
+            return None
+
+    @property
+    def intleaks(self):
+        '''Retrieve interior :ref:``Leak`` class array from a :ref:``Photon`` class '''
+        if self.photon is NULL:
+            return None
+
+        cdef polycap_error *error = NULL
+
+        #logger.debug('Before calling C')
+        if self.n_int_leaks == 0:
+            polycap_photon_get_intleak_data(self.photon, &self.int_leaks, &self.n_int_leaks, &error)
+            polycap_set_exception(error)
+        #logger.debug('After calling C') 
+
+        rv = list()
+
+        if self.n_int_leaks > 0:
+            for i in range(self.n_int_leaks):
+                leak = Leak.create(self.int_leaks[i])
+                yield leak
+        else:
+            return None
+
     def get_exit_coords(self):
         '''Retrieve exit coordinates from a :ref:``Photon`` class'''
         return vector2tuple(polycap_photon_get_exit_coords(self.photon))
 
+    @property
+    def exit_coords(self):
```

Comment:
Wees ook consistent met gebruik van underscores in method names, in `TransmissionEfficiencies` is er geen underscore hier voor dezelfde method.

Reply:
Idd, omdat er anders verwarring was met de member variable ;) Maar dat zal nu dus fixed worden daar member variable `_exit_coords` wordt, en de property dus `exit_coords` :P